[BUG] React error when first RPC provider fails in parallel strategy

## Description

When using the parallel RPC strategy and the first configured RPC provider fails, the application crashes with a React error when attempting to render transaction/block/address data. The error indicates that React is trying to render a raw `RPCProviderResponse` object instead of the properly transformed domain object.

## Steps to Reproduce

1. Configure multiple RPC providers for a network (e.g., Ethereum mainnet)
2. Enable parallel RPC strategy in Settings
3. Ensure the first RPC provider in the list will fail (either misconfigured URL or provider is down)
4. Navigate to a transaction, block, or address page
5. Observe the React error

## Expected Behavior

When the first RPC provider fails:
- The application should fall back to subsequent successful providers
- The data should be properly extracted and transformed using the adapter layer
- The UI should render correctly with data from the first successful provider
- The RPC indicator should show which providers failed and which succeeded

## Actual Behavior

The application crashes with the following React error:

```
Error: Objects are not valid as a React child (found: object with keys {url, status, responseTime, error}).
If you meant to render a collection of children, use an array instead.
```

Stack trace in production mode:
```
installHook.js:1 Error: Objects are not valid as a React child (found: object with keys {url, status, responseTime, error}).
If you meant to render a collection of children, use an array instead.
    at throwOnInvalidObjectType (react-dom-client.development.js:7081:13)
    at createChild (react-dom-client.development.js:7359:11)
    at reconcileChildrenArray (react-dom-client.development.js:7628:25)
    at reconcileChildFibersImpl (react-dom-client.development.js:7951:30)
    at eval (react-dom-client.development.js:8056:33)
    at reconcileChildren (react-dom-client.development.js:8620:13)
    at beginWork (react-dom-client.development.js:10792:13)
    at runWithFiberInDEV (react-dom-client.development.js:1519:30)
    at performUnitOfWork (react-dom-client.development.js:15131:22)
    at workLoopSync (react-dom-client.development.js:14955:41)

The above error occurred in the <span> component.
```

Dev mode error:
```
ErrorBoundary caught an error: Error: Objects are not valid as a React child (found: object with keys {url, status, responseTime, error}).
If you meant to render a collection of children, use an array instead.
    at throwOnInvalidObjectType (react-dom-client.development.js:7081:13)
    at createChild (react-dom-client.development.js:7359:11)
    at reconcileChildrenArray (react-dom-client.development.js:7628:25)
    at reconcileChildFibersImpl (react-dom-client.development.js:7951:30)
    at eval (react-dom-client.development.js:8056:33)
    at reconcileChildren (react-dom-client.development.js:8620:13)
    at beginWork (react-dom-client.development.js:10792:13)
    at runWithFiberInDEV (react-dom-client.development.js:1519:30)
    at performUnitOfWork (react-dom-client.development.js:15131:22)
    at workLoopSync (react-dom-client.development.js:14955:41)
```

## Root Cause Analysis

The issue occurs in the `useSelectedData` hook ([src/hooks/useSelectedData.ts:43](src/hooks/useSelectedData.ts#L43)):

```typescript
// Return the selected provider's data
return providerResponse.data;
```

### Problem Flow:

1. **Parallel Strategy Returns Raw Responses**: In parallel mode, `ParallelStrategy.execute()` returns the full array of `RPCProviderResponse` objects as `result.data` ([explorer-network-connectors/dist/strategies/parallelStrategy.js:118](node_modules/explorer-network-connectors/dist/strategies/parallelStrategy.js#L118))

2. **Adapter Layer Extracts Default Data**: The `extractData()` helper ([src/services/adapters/shared/extractData.ts:10-35](src/services/adapters/shared/extractData.ts#L10-L35)) correctly extracts the first successful response's `.data` field and assigns it to `result.data`

3. **Metadata Preserves Raw Responses**: The `result.metadata.responses` array contains the original `RPCProviderResponse` objects with their raw `.data` fields (which are RPC responses, not domain objects)

4. **Provider Selection Bug**: When a user switches providers or when `useSelectedData` tries to use a specific provider, it accesses `providerResponse.data` from `metadata.responses`, which contains the **raw RPC response** before adapter transformation, not the transformed domain object (Block, Transaction, Address, etc.)

5. **React Rendering Fails**: The raw response object (with keys like `url`, `status`, `responseTime`, `error`) gets passed to a component that tries to render it, causing React to throw the "Objects are not valid as a React child" error

### The Data Inconsistency:

```typescript
// After parallel strategy execution:
result.data = extractData(responses) // ✓ Correctly extracted domain object (Transaction, Block, etc.)
result.metadata.responses[0].data = rawRpcResponse // ✗ Raw RPC response (not transformed)

// When user selects a provider:
useSelectedData returns providerResponse.data // ✗ Returns raw RPC response, not domain object
```

## Technical Details

### Affected Files:

1. **[src/hooks/useSelectedData.ts:32-43](src/hooks/useSelectedData.ts#L32-L43)** - Returns raw provider response data instead of transformed data
2. **[src/services/adapters/shared/extractData.ts](src/services/adapters/shared/extractData.ts)** - Extracts data correctly for default provider but doesn't help with provider switching
3. **[node_modules/explorer-network-connectors/dist/strategies/parallelStrategy.js:112-130](node_modules/explorer-network-connectors/dist/strategies/parallelStrategy.js#L112-L130)** - Stores raw RPC responses in metadata

### Data Flow:

```
ParallelStrategy.execute()
  ↓ Returns responses array with raw RPC data
NetworkAdapter.getTransaction/getBlock/getAddress()
  ↓ Uses extractData() to get first successful response's raw data
  ↓ Transforms raw data → domain object (Transaction, Block, Address)
  ↓ Returns { data: domainObject, metadata: { responses: [...rawResponses] } }
Component (via useSelectedData)
  ↓ result.data = domainObject ✓
  ↓ result.metadata.responses[i].data = rawRpcResponse ✗
  ↓ When selecting provider: returns rawRpcResponse
React tries to render object
  ↓ ERROR: Objects are not valid as a React child
```

## Environment

- **Browser**: All browsers
- **OS**: All operating systems
- **Network**: All networks when using parallel RPC strategy
- **RPC Strategy**: Parallel (fallback strategy is not affected)

## Additional Context

- This bug only affects **parallel RPC strategy**. Fallback mode works correctly because it doesn't use `metadata.responses`.
- The error is caught by the ErrorBoundary component, preventing complete application crash
- Users can work around this by:
  - Switching back to fallback RPC strategy
  - Removing failed RPC providers from configuration
  - Ensuring the first RPC provider in the list is always available

## Related Code References

- [src/types/index.ts:294-312](src/types/index.ts#L294-L312) - `RPCMetadata` and `RPCProviderResponse` type definitions
- [src/hooks/useSelectedData.ts](src/hooks/useSelectedData.ts) - Hook that returns incorrect data
- [src/services/adapters/shared/extractData.ts](src/services/adapters/shared/extractData.ts) - Data extraction helper
- [src/components/common/RPCIndicator.tsx:71-105](src/components/common/RPCIndicator.tsx#L71-L105) - Maps through provider responses
- [node_modules/explorer-network-connectors/dist/strategies/parallelStrategy.js](node_modules/explorer-network-connectors/dist/strategies/parallelStrategy.js) - Parallel strategy implementation

--- a/src/services/adapters/shared/extractData.ts
+++ b/src/services/adapters/shared/extractData.ts
@@ -13,13 +13,8 @@ export function extractData<T>(strategyResultData: T | any): T {
     const firstItem = strategyResultData[0];
 
     // Check if it has the RPCProviderResponse structure
-    if (
-      firstItem &&
-      typeof firstItem === "object" &&
-      "url" in firstItem &&
-      "status" in firstItem &&
-      "data" in firstItem
-    ) {
+    // Note: error responses may not have 'data' field, so we check for url+status
+    if (firstItem && typeof firstItem === "object" && "url" in firstItem && "status" in firstItem) {
       // Find the first successful response
       const successfulResponse = strategyResultData.find(
         // biome-ignore lint/suspicious/noExplicitAny: Provider response type is dynamic

